packages/page-accounts/src/Accounts/index.spec.ts

```diff
@@ -299,6 +301,178 @@ describe('Accounts page', () => {
+          renderAccountsWithDefaultAddresses(
+            anAccountWithInfo({ flags: { isEditable: false } as AddressFlags })
+          );
+          sideBar = await openSidebarForAccountRow(0);
+        });
+
+        describe('if edit button has not been pressed', () => {
+          it('tags', async () => {
+            await sideBar.clickByText('no tags');
+            expect(sideBar.queryByRole('combobox')).toBeFalsy();
+          });
+
+          it('account name', async () => {
+            const inputNotFoundError = 'Unable to find an element by: [data-testid="name-input"]';
+
+            await expect(sideBar.typeAccountName(newName)).rejects.toThrowError(inputNotFoundError);
+          });
+        });
+      });
+
+      describe('on edit cancel', () => {
+        beforeEach(async () => {
+          addAccountToKeyring(injectedAddress, { isDevelopment: false, name: initialName, tags: [] });
+          renderAccountsForAddresses(
+            injectedAddress
+          );
+          sideBar = await openSidebarForAccountRow(0);
+          await sideBar.assertTags('no tags');
+          sideBar.edit();
+        });
+
+        it('restores tags and name to state from keyring', async () => {
+          await sideBar.typeAccountName(newName);
+          await sideBar.selectTag(defaultTag);
+
+          sideBar.cancel();
+          await sideBar.assertTags('no tags');
+          await sideBar.assertAccountName(initialName);
+        });
+
+        it('Cancel button disappears', () => {
+          sideBar.cancel();
+          expect(sideBar.queryByRole('button', { name: 'Cancel' })).toBeFalsy();
+        });
+      });
+
+      describe('outside click', () => {
+        beforeEach(async () => {
+          renderAccountsForAddresses(
+            aliceAddress,
+            bobAddress
+          );
+
+          sideBar = await openSidebarForAccountRow(0);
+          sideBar.edit();
+        });
+
+        it('cancels editing', async () => {
+          await sideBar.typeAccountName(newName);
+          await sideBar.selectTag(defaultTag);
+
+          fireEvent.click(await screen.findByText('accounts'));
+
+          await sideBar.assertTags('no tags');
+          await sideBar.assertAccountName('ALICE');
+
+          expect(sideBar.queryByRole('button', { name: 'Cancel' })).toBeFalsy();
+        });
+
+        it('within sidebar does not cancel editing', async () => {
+          await sideBar.clickByText('Tags');
+
+          expect(sideBar.queryByRole('button', { name: 'Cancel' })).toBeTruthy();
+        });
+
+        it('cancels editing and changes name when opening sidebar for another account', async () => {
+          await waitFor(() => sideBar.accountNameEquals('ALICE'));
+
+          sideBar = await openSidebarForAccountRow(1);
+          await sideBar.assertAccountName('BOB');
+        });
+      });
+
+      afterEach(() => {
+        keyring.forgetAccount(injectedAddress);
+        keyring.forgetAddress(injectedAddress);
```

Comment:
Those lines trigger a `useEffect`, that updates some react components. This raises a warning in tests, as we don't tell react we're expecting it. We should either (1) call [cleanup](https://testing-library.com/docs/react-testing-library/api#cleanup) before those lines or (2) wrap them in `act` to tell react component updates are OK here.